node/src/transition_frontier/sync/transition_frontier_sync_effects.rs

```diff
@@ -57,6 +58,17 @@ impl TransitionFrontierSyncAction {
                 // TODO(binier): cleanup ledgers
             }
             TransitionFrontierSyncAction::LedgerStakingPending => {
+                // The staking ledger is equal to the genesis ledger with changes on top, so
+                // we use it as a base to save work during synchronization.
+                let best_tip = store.state().transition_frontier.sync.best_tip().unwrap();
+                let target = super::ledger::SyncLedgerTarget::staking_epoch(best_tip);
+                let origin = best_tip.genesis_ledger_hash().clone();
```

Comment:
For very unlikely edge case.
If we have any ledgers for `store.state().transition_frontier.best_tip()` (already synced best tip), use that instead of genesis.

Reply:
@binier can you expand on this one? note sure I follow.